internal/service/searchindex/resource_search_index.go

```diff
@@ -320,15 +319,39 @@ func resourceMongoDBAtlasSearchIndexRead(ctx context.Context, d *schema.Resource
 		return diag.Errorf("error setting `analyzer` for search index (%s): %s", d.Id(), err)
 	}
 
-	if analyzers := searchIndex.GetAnalyzers(); len(analyzers) > 0 {
+	if searchIndex.Type != nil && *searchIndex.Type == vectorSearch {
```

Comment:
q: if these values are coming from the API, is there a need to add this conditional depending on search index type?

Reply:
yes, because depending on the type only some fields are allowed to be filled

Reply:
But for this case that it is processing the response from the api into the terraform state it would not be necessary right?

Reply:
it is because now the marshals now return empty slices (that are not nil pointers) so if we don't do this there will be an error because an unexpected change plan